### PR TITLE
App Engine - A series of Generator Fixes

### PR DESCRIPTION
**Pull request by saturnism**
- Re-introduce class loading priority; this is needed for specifically echcache, and adding the mail api just in case.
- Add initial Gradle support
- Store database name in the right variable for default value
- Configure UTF-8 encoding via web.xml for static files

For #7750

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by saturnism** [comment]
Alright I think that's the last set of updates for this PR :)

File: generators/gae/templates/appengine-web.xml.ejs [hunk 1]
```diff
@@ -30,6 +30,13 @@
     <system-properties>
         <property name="java.util.logging.config.file" value="WEB-INF/logging.properties"/>
     </system-properties>
+    <class-loader-config>
+        <!-- temporary workaround to make sure the app uses it's memcache/mail jars.!-->
+        <priority-specifier filename="cache-api-1.0.0.jar"/>
+        <priority-specifier filename="cache-api-1.1.0.jar"/>
+        <priority-specifier filename="javax-mail-1.5.6.jar"/>
+        <priority-specifier filename="javax.mail-1.6.2.jar"/>
```
Comment (ludoch) [inline generators/gae/templates/appengine-web.xml.ejs]: LGTM..
Curious: 
1/ we were not doing that before?
2/ do we really need to have the 2 versions? If folks use the latest jHispter, they should only have the new cache and mail jars, so 1 of them would only be needed?

Comment (saturnism) [reply generators/gae/templates/appengine-web.xml.ejs]: we had these before, but i removed it while testing w/ hazelcast thought we didn't need these anymore. my mistake :)

i don't think we need the other versions; removing them from this list.

**Comment by ludoch** [comment]
LGTM

**Comment by deepu105** [comment]
@saturnism is this ready to be merged?

**Comment by sendilkumarn** [comment]
@deepu105 I am testing it. Will merge it once done 👍

**Comment by sendilkumarn** [comment]
Is it failing because of the flaky protractor?

**Comment by saturnism** [comment]
looks like all green :D Thanks @sendilkumarn !

**Pull request merged**